test/language/generators/syntax/return.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2013 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+  description: >
+      `return` is a valid statement within generator function bodies.
+  es6id: 14.4
+ ---*/
+
+function* g() { return; }
+function* g() { return 1; }
+function* g() {
```

Comment:
I suppose to account for an edge case where a newline would trigger different parsing behavior. That seems more like a regression test, though. If we were to try to catch all problems like this exhaustively, we'd be here all day! I'll remove all the duplicates you've identified.